subsys/net/lib/nrf_cloud/src/nrf_cloud_info.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <net/nrf_cloud.h>
+#if defined(CONFIG_MODEM_INFO)
+#include <modem/modem_info.h>
+#endif
+#if defined(CONFIG_MODEM_JWT)
+#include <modem/modem_jwt.h>
+#endif
+#include "nrf_cloud_transport.h"
+#include "nrf_cloud_client_id.h"
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(nrf_cloud_info, CONFIG_NRF_CLOUD_LOG_LEVEL);
+
+#if defined(CONFIG_MODEM_INFO) && defined(CONFIG_NRF_CLOUD_VERBOSE_DETAILS)
+static struct modem_param_info mdm_param;
```

Comment:
i know we already use `modem_info` elsewhere in the `nrf_cloud` library, but the 2 uses of `modem_info_string_get` could be replaced with direct calls to `nrf_modem_at_cmd`.

and you could have a single at cmd buffer instead of the huge (2552 bytes) `struct modem_param_info`.

Reply:
also you could expose separate functions for getting IMEI and mfw ver, which could be used by `nrfc_dtls.c` and `nrf_cloud_client_id.c`.

the `struct modem_param_info` could then be removed from `nrfc_dtls.c`

Reply:
good points

Reply:
Actually, IMEI and mfw ver are not used in `nrfc_dtls.c`.

I will remove the `struct modem_param_info` and just read the IP address string directly.

`IMEI` is used in `nrf_cloud_client_id.c` but the savings would be minimal.